src/module/helpers.ts

```diff
@@ -400,9 +401,59 @@ export class Helpers {
         damage.element.base = element;
         damage.element.value = element;
 
+        if (sourceItem && sourceItem.actor) {
+            damage.source = {
+                actorId: sourceItem.actor.id,
+                itemType: sourceItem.type,
+                itemId: sourceItem.id,
+                itemName: sourceItem.name
+            };
+        }
+
         return damage;
     }
 
+    /**
+     * Retrieves the item causing the damage, if there is any. 
+     * This only works for embedded items at the moment
+     */
+    static findDamageSource(damageData : DamageData) : SR5Item | undefined{
+        if (!damageData.source) {
+            return;
+        }
+        
+        const actorId = damageData.source.actorId;
+        const actorSource = game.actors.find(
+            actor => actor.id === actorId
+            );
+
+        if (!actorSource) {
+            return;
+        }
+
+        // First search the actor itself for the item
+        const itemId = damageData.source.itemId;
+        const actorItem = actorSource.getOwnedItem(itemId) as SR5Item;
+        if (actorItem)
+        {
+            return actorItem;
+        }
+
+        // If we did not find anything on the actor, search the active tokens (the item might only exist on a non linked token)
+        // This will not work if we are on a different scene or the token got deleted, which is expected when you put an
+        // item on a token without linking it.
+        const tokens = actorSource.getActiveTokens();
```

Comment:
This will only work if the token match the active scene. However I think this is okay, since at some point a closed scene should not be valid anymore (and also might not contain the causing token+item anymore). For opposed testes there has been a sceneId.tokenId combination for that reason. But again, don't think it's a big issue and I'd rather let that 'feature' slide out, since it's very obtuse to the user that an opened item would come from some token on some scene...